src/js_native_api_v8.cc

```diff
@@ -2526,15 +2526,16 @@ napi_status NAPI_CDECL napi_get_value_string_utf16(napi_env env,
     // V8 assumes UTF-16 length is the same as the number of characters.
     *result = val.As<v8::String>()->Length();
   } else if (bufsize != 0) {
-    int copied = val.As<v8::String>()->Write(env->isolate,
-                                             reinterpret_cast<uint16_t*>(buf),
-                                             0,
-                                             bufsize - 1,
-                                             v8::String::NO_NULL_TERMINATION);
+    size_t n_chars = std::min(
+        bufsize - 1, static_cast<size_t>(val.As<v8::String>()->Length()));
```

Comment:
It would be nice to avoid calling `val.As<v8::String>()` twice.
In case if it does additional checks then it may cause unnecessary overhead.

Reply:
it only has checks in debug mode: https://github.com/nodejs/node/blob/264cad75ce08dabb6ddc8cc56cc70c9ba3447447/deps/v8/include/v8-local-handle.h#L324-L332

Reply:
Great! It is still better to avoid calling the same code twice. :)

Reply:
FWIW, I originally decided against this because it'd be inconsistent with other functions in this file, but I suppose I might as well.